XSIAM/xsiam_modules/convert_xsiam.py

```diff
@@ -0,0 +1,149 @@
+from typing import Any
+from pydantic.v1 import BaseModel
+from sekoia_automation.action import Action
+import re
+
+from xsiam_modules.helpers import iso8601_to_timestamp
+
+TYPE_MAPPING = {
+    "file": "HASH",
+    "filename": "FILENAME",
+    "directory": "PATH",
+    "domain-name": "DOMAIN_NAME",
+    "url": "DOMAIN_NAME",
+    "ipv4-addr": "IP",
+    "ipv6-addr": "IP",
+    "network-traffic": "IP",
+}
+
+HIGH_KILL_CHAIN_PHASES = ["exploitation", "installation", "command-and-control", "actions-on-objectives"]
+
+
+class ActionArguments(BaseModel):
+    objects: list[dict]  # List of Stix objects to convert
+
+
+class ConvertSTIXToXSIAMAction(Action):
+    def run(self, arguments: ActionArguments) -> Any:
+        """
+        Convert STIX objects to XSIAM format.
+        """
+        xsiam_objects = []
+
+        for stix_obj in arguments.objects:
+            patterns = self._get_pattern(stix_obj)
+
+            for pattern in patterns:
+                xsiam_obj = self._create_xsiam_object(pattern, stix_obj)
+                xsiam_objects.append(xsiam_obj)
+
+        return xsiam_objects
+
+    def _create_xsiam_object(self, pattern: str, stix_obj: dict) -> dict:
+        return {
+            "indicator": pattern,
+            "type": self._get_type(stix_obj),
+            "severity": self._get_severity(stix_obj),
+            "expiration_date": iso8601_to_timestamp(stix_obj.get("valid_until", "1970-01-01T00:00:00Z")),
+            "comment": self._get_comment(stix_obj),
+            "reputation": self._get_reputation(stix_obj),
+            "reliability": self._get_reliability(stix_obj),
+            "vendors": self._get_vendors(stix_obj),
+            "class": stix_obj.get("id", "N/A"),
+            "validate": True,
+        }
+
+    def _get_pattern(self, stix_obj: dict) -> list[str]:
+        """
+        Convert the pattern from STIX to XSIAM format.
+        """
+        pattern = stix_obj.get("pattern")
+        if pattern is None:
+            return ["No valid ioc found"]
+
+        if hashes:=re.findall(r"file:hashes\.'?(?:MD5|SHA-256)'?\s?=\s?'(?P<value>.*?)'", pattern, flags=re.DOTALL | re.MULTILINE):
+            return hashes
+        else:
+            return re.findall(r":value\s?=\s?'(?P<value>.+?)'", pattern, flags=re.DOTALL | re.MULTILINE)
```

Comment:
Why only focus on MD5 or SHA256 ? I think a generic `=\s?'(?P<value>.+?)'` should cover all cases

Reply:
Yes according to the playbook, we dont care about the SHA1 for example but we can make it generic